src/olympia/lib/git.py

```diff
@@ -246,6 +249,23 @@ def __init__(self, addon_or_id, package_type='addon'):
     def is_extracted(self):
         return os.path.exists(self.git_repository_path)
 
+    @property
+    def is_recent(self):
+        if not self.is_extracted:
+            return False
+        # A git repository is recent when it was created less than 1 hour ago.
+        an_hour_ago = datetime.utcnow() - timedelta(hours=1)
```

Comment:
so, I agree this is a good idea but the end result is we'll still be attempting extraction 24 times a day for a perpetually broken git repo when we may not look at the errors for days/weeks.  Maybe the delay should be longer?  12 hours rather than 1?

Reply:
I think we'll see how it goes. We have statsd metrics for that.

I am not sure we would end up in a situation where an add-on cannot be re-extracted at all for 24h. If we loop as you describe, it means new versions are also uploaded during that time.

Otherwise, if we have an extraction problem on a new repo, we won't re-extract the repo anymore (until a new version gets submitted).